### ISSUE TITLE
Enhance the MetricsCollector to distinguish metrics for NACK (requeue = true) vs NACK (requeue = false)

### ISSUE THREAD

**Issue by matsev** [rabbitmq/rabbitmq-java-client#1442]
### Is your feature request related to a problem? Please describe.

To my understanding there are currently no metrics emitted that distinguish whether or not a message was requeued when it was NACK:ed by the consumer

### Describe the solution you'd like

One suggestion is to enhance the [MetricsCollector](https://github.com/rabbitmq/rabbitmq-java-client/blob/main/src/main/java/com/rabbitmq/client/MetricsCollector.java) with a new methods:

```java
    default public void basicNack(Channel channel, long deliveryTag, boolean requeue) {
    }

    default public void basicReject(Channel channel, long deliveryTag, boolean requeue) {
    }
```

These methods would then be called with the `requeue` parameter provided whenever a messages is NACK:ed. Moreover, existing implementation such as the [MicrometerMetricsCollector](https://github.com/rabbitmq/rabbitmq-java-client/blob/main/src/main/java/com/rabbitmq/client/impl/MicrometerMetricsCollector.java) should implement the methods so that the metrics emitted also has knowledge about the `requeue` state

### Describe alternatives you've considered

Application developers can implement their own custom metrics

### Additional context

Related question at Stack Overflow: https://stackoverflow.com/questions/79069702/reactive-rabbitmq-metrics-for-ack-nack-requeue-false-and-nack-requeue-t

**Comment by acogoluegnes** [comment]
This makes sense. The tricky part would be to maintain compatibility by keeping the nacked and rejected existing metrics _without_ the requeue information, and new metrics with the requeue information.

PR welcome.

**Comment by ohksj77** [comment]
Fixed PR: #1469

**Comment by matsev** [comment]
I did a quick test and I can verify that the `requeued` metrics is emitted by the `MicrometerMetricsCollector`. Thanks!
